src/index.js

```diff
@@ -1,28 +1,46 @@
 import { isAbsolute, join } from 'path';
 
 import arrify from 'arrify';
+import micromatch from 'micromatch';
 
 import { getOptions } from './options';
-import DirtyFileWatcher from './DirtyFileWatcher';
 import linter from './linter';
-import { parseFiles } from './utils';
+import { parseFiles, parseFoldersToGlobs } from './utils';
 
 /** @typedef {import('webpack').Compiler} Compiler */
 /** @typedef {import('./options').Options} Options */
 
+const ESLINT_PLUGIN = 'ESLintWebpackPlugin';
+
 class ESLintWebpackPlugin {
   /**
    * @param {Options} options
    */
   constructor(options = {}) {
     this.options = getOptions(options);
+    this.run = this.run.bind(this);
   }
 
   /**
    * @param {Compiler} compiler
    * @returns {void}
    */
   apply(compiler) {
+    if (!this.options.lintDirtyModulesOnly) {
+      compiler.hooks.run.tapPromise(ESLINT_PLUGIN, this.run);
+    }
+
+    // TODO: Figure out want `compiler.watching` is and how to use it in Webpack5.
+    // From my testing of compiler.watch() ... compiler.watching is always
+    // undefined (webpack 4 doesn't define it either) I'm leaving it out
+    // for now.
+    compiler.hooks.watchRun.tapPromise(ESLINT_PLUGIN, this.run);
```

Comment:
/cc @ricardogobbosouza What do you think about improving it for auto detecting watch only on webpack@5?

Reply:
@evilebottnawi Sorry, but I didn't understand your question lol